discord-scripts/thread-management/auto-join.ts

```diff
@@ -39,17 +43,16 @@ async function autoJoinThread(
   const placeholder = await thread.send("<placeholder>")
 
   // Use this to assign a specific role based on the mapping in CUSTOM_CHANNEL_ROLE, in order to map specific roles/channels
-  const matchingChannel = CUSTOM_CHANNEL_ROLE.find(
-    (channel) => containingChannel?.name.endsWith(channel.channelName),
-  )
-  if (matchingChannel) {
-    const channelMatchingRole = server.roles.cache.find(
-      (role) =>
-        role.name.toLowerCase() === matchingChannel.roleName.toLowerCase(),
-    )
-    if (channelMatchingRole) {
-      await placeholder.edit(channelMatchingRole.toString())
-      return
+  if (hasCustomChannels && containingChannel) {
```

Comment:
The way this control flows, we don't even need `hasCustomChannels`—if there are no custom channels, `roleName` will never pass `if (roleName)`, and we'll continue to the following components.